app/forms/declaration/additionaldocuments/DocumentWriteOff.scala

```diff
@@ -16,44 +16,79 @@
 
 package forms.declaration.additionaldocuments
 
-import play.api.data.Forms.{bigDecimal, optional, text}
-import play.api.data.{Form, Forms}
+import play.api.data.Forms.{optional, text}
+import play.api.data.{Form, FormError, Forms}
 import play.api.libs.json.Json
-import utils.validators.forms.FieldValidator.{isAlphanumericWithSpecialCharacters, lengthInRange}
+import utils.validators.forms.FieldValidator._
 
 case class DocumentWriteOff(measurementUnit: Option[String], documentQuantity: Option[BigDecimal])
 
 object DocumentWriteOff {
 
+  def convert(measurementUnit: Option[String], documentQuantity: Option[String]): DocumentWriteOff =
+    new DocumentWriteOff(measurementUnit, documentQuantity.map(BigDecimal(_)))
+
   implicit val format = Json.format[DocumentWriteOff]
 
-  private val measurementUnitMinLength = 3
-  private val measurementUnitMaxLength = 5
+  private val measurementUnitLength = 3
+  private val qualifierLength = 1
   private val documentQuantityMaxLength = 16
   private val documentQuantityMaxDecimalPlaces = 6
 
+  val documentWriteOffKey = "documentWriteOff"
   val measurementUnitKey = "measurementUnit"
+  val qualifierKey = "qualifier"
   val documentQuantityKey = "documentQuantity"
 
+  private def form2Model: (Option[String], Option[String], Option[String]) => DocumentWriteOff = {
+    case (None, None, quantity)      => convert(None, quantity)
+    case (unit, None, quantity)      => convert(unit, quantity)
```

Comment:
those two cases can be in one case. `case (unit, None, quantity)      => convert(unit, quantity)` this is enough, if unit will be None you will pass unit to the convert method, if not None

Reply:
Good spot